### PR TITLE
Add python shared memory string client examples

### PR DESCRIPTION
**Pull request by CoderHam**
- [x] Add `simple_cuda_shm_string_client.py`
- [x] Add `simple_cuda_shm_string_client.py`
- [x] Update tests for _L0_simple_shared_memory_example_ and _L0_simple_cuda_shared_memory_example_

### CODE REVIEW

File: src/clients/python/cuda_shared_memory/__init__.py [hunk 1]
```diff
@@ -73,6 +73,54 @@ def _raise_error(msg):
     ex = CudaSharedMemoryException(msg)
     raise ex
 
+def serialize_string_tensor(input_tensor):
+    """Creates a shared memory region with the specified name and size.
```
Comment (deadeyegoodwin) [inline src/clients/python/cuda_shared_memory/__init__.py]: This comment is wrong

File: src/clients/python/cuda_shared_memory/__init__.py [hunk 2]
```diff
@@ -73,6 +73,54 @@ def _raise_error(msg):
...
+
+    Parameters
+    ----------
+    input_tensor : np.array
+        The string tensor to serialize.
+
+    Returns
+    -------
+    serialized_string_tensor : np.array
+        The serialized string tensor.
```
Comment (deadeyegoodwin) [inline src/clients/python/cuda_shared_memory/__init__.py]: Say something about how the serialization is a tensor of type uint8 (or char)

File: src/clients/python/shared_memory/__init__.py [hunk 3]
```diff
@@ -73,6 +73,55 @@ def _raise_error(msg):
     ex = SharedMemoryException(msg)
     raise ex
 
+def serialize_string_tensor(input_tensor):
+    """Creates a shared memory region with the specified name and size.
```
Comment (deadeyegoodwin) [inline src/clients/python/shared_memory/__init__.py]: Fix

File: src/clients/python/shared_memory/__init__.py [hunk 4]
```diff
@@ -73,6 +73,55 @@ def _raise_error(msg):
...
+
+    Parameters
+    ----------
+    input_tensor : np.array
+        The string tensor to serialize.
+
+    Returns
+    -------
+    serialized_string_tensor : np.array
+        The serialized string tensor.
```
Comment (deadeyegoodwin) [inline src/clients/python/shared_memory/__init__.py]: Same fix as above

File: src/clients/python/cuda_shared_memory/__init__.py (hunk 2 above)
Comment (CoderHam) [reply src/clients/python/cuda_shared_memory/__init__.py]: Done

File: src/clients/python/cuda_shared_memory/__init__.py (hunk 1 above)
Comment (CoderHam) [reply src/clients/python/cuda_shared_memory/__init__.py]: Fixed

File: src/clients/python/shared_memory/__init__.py (hunk 4 above)
Comment (CoderHam) [reply src/clients/python/shared_memory/__init__.py]: Done

File: src/clients/python/shared_memory/__init__.py (hunk 3 above)
Comment (CoderHam) [reply src/clients/python/shared_memory/__init__.py]: Done

**Comment by CoderHam** [comment]
> Looks ok but CI failed for both new examples so something is wrong somewhere

The failure was caused because I missed adding these shared memory clients in the cmake file

**Comment by CoderHam** [comment]
> Looks ok but CI failed for both new examples so something is wrong somewhere

I had missed adding these in the cmake and they did not get copied successfully. Tested locally, re-ran CI for the same

**Pull request merged**